lib/pay/env.rb

```diff
@@ -40,11 +38,15 @@ def env
     end
 
     def secrets
-      Rails.application.secrets
+      is_rails_newer_or_equal?(720) ? nil : Rails.application.secrets
     end
 
     def credentials
       Rails.application.credentials if Rails.application.respond_to?(:credentials)
     end
+
+    def is_rails_newer_or_equal?(version)
+      Rails.version.delete(".").to_i >= version
```

Comment:
@excid3 : after I've pushed this code I've realized that Rails version 7.1.2.1 will break this logic, so I changed the PR to `Draft` and will work on that later.

And also remembered this I saw this in the code, while looking around `Rails.gem_version >= Gem::Version.new("6.1")`.